java/core/src/main/java/com/microsoft/bond/NullableBondType.java

```diff
@@ -82,6 +82,11 @@ protected final TValue newDefaultValue() {
     }
 
     @Override
+    protected final TValue cloneValue(TValue value) {
+        return value == null ? null : this.valueType.cloneValue(value);
```

Comment:
Cloning doesn't check for this condition (serialization does). It may be cloned as is, and an error will manifest during serialization. Or it may fail with null reference, which we can't always detect without significant perf penalty.

---
In reply to: [128379441](https://github.com/Microsoft/bond/pull/532#discussion_r128379441) [](ancestors = 128379441)